Arquivo final 3 sprint


In [ ]:
import pandas as pd

In [ ]:
data = {'coluna_1':[1.5,2.0,1.7,2.2,1.9],
        'coluna_2':[10.0,12.0,11.5,13.0,11.0],
        'coluna_3':[0,1,0,1,0],
        'coluna_4':[1,0,1,1,0],
        'coluna_5':[20.5,25.5,22.0,27.0,23.0]}

In [ ]:
df = pd.DataFrame(data)

In [ ]:
print(df)

   coluna_1  coluna_2  coluna_3  coluna_4  coluna_5
0       1.5      10.0         0         1      20.5
1       2.0      12.0         1         0      25.5
2       1.7      11.5         0         1      22.0
3       2.2      13.0         1         1      27.0
4       1.9      11.0         0         0      23.0


In [ ]:
from google.colab import drive


In [ ]:
drive.mount('/content/drive')


Mounted at /content/drive


In [ ]:
origem = '/content/drive/MyDrive/Data/COMEXSTAT/'

In [ ]:
arquivo_1 = origem + 'exp.csv'
arquivo_2 = origem + 'imp.csv'

In [ ]:
exp = pd.read_csv(arquivo_1, low_memory=False, sep=',', encoding='UTF-8', quotechar='"')
imp = pd.read_csv(arquivo_2, low_memory=False, sep=',', encoding='UTF-8', quotechar='"')

In [ ]:
dado_final.head(10)

,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao,Saldo_Balança_Comercial
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,7788.0,0.0,966.0,0.0,7788.0
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,1544.0,0.0,1.0,0.0,1544.0
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,907.0,0.0,73.0,0.0,907.0
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,3942.0,0.0,75.0,0.0,3942.0
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,29604.0,0.0,6949.0,0.0,29604.0
5,2021,7,1604,434,PR,4118204,111,115,1604,Preparações e conservas de peixes; caviar e se...,Libéria,4118204,Paranaguá,Export,115.0,0.0,111.0,0.0,115.0
6,2021,12,9404,386,MG,3110301,105,3045,9404,"Suportes elásticos para camas; colchões, edred...",Itália,3110301,Caldas,Export,3045.0,0.0,105.0,0.0,3045.0
7,2021,9,901,87,MG,3103504,79125,201612,901,"Café, mesmo torrado ou descafeinado; cascas e ...",Bélgica,3103504,Araguari,Export,201612.0,0.0,79125.0,0.0,201612.0
8,2021,10,408,538,SP,3448500,185,540,408,"Ovos de aves, sem casca, e gemas de ovos, fres...",Noruega,3448500,Santos,Export,540.0,0.0,185.0,0.0,540.0
9,2021,10,8517,275,SP,3450308,9,814,8517,Aparelhos elétricos para telefonia ou telegraf...,França,3450308,São Paulo,Export,814.0,0.0,9.0,0.0,814.0


In [ ]:
# Re-initialize and populate the columns to ensure they exist
dado_final['VL_FOB_Exportacao'] = 0.0
dado_final['VL_FOB_Importacao'] = 0.0
dado_final['KG_LIQUIDO_Exportacao'] = 0.0
dado_final['KG_LIQUIDO_Importacao'] = 0.0

dado_final.loc[dado_final['Tipo_Operacao'] == 'Export', 'VL_FOB_Exportacao'] = dado_final['VL_FOB']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Export', 'KG_LIQUIDO_Exportacao'] = dado_final['KG_LIQUIDO']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Import', 'VL_FOB_Importacao'] = dado_final['VL_FOB']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Import', 'KG_LIQUIDO_Importacao'] = dado_final['KG_LIQUIDO']

dado_final['Saldo_Balança_Comercial'] = dado_final['VL_FOB_Exportacao'] - dado_final['VL_FOB_Importacao']

# Now attempt the type conversion for the value columns
value_columns = ['VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', 'KG_LIQUIDO_Importacao', 'Saldo_Balança_Comercial']
for col in value_columns:
    # Ensure the column exists before trying to convert, and handle potential NaN values
    if col in dado_final.columns:
        dado_final[col] = dado_final[col].astype(float).fillna(0).astype(int)
    else:
        print(f"Warning: Column '{col}' not found in dado_final. Skipping type conversion for this column.")

print("Decimal points removed from value columns. Displaying head of the updated DataFrame:")
dado_final.head()

Decimal points removed from value columns. Displaying head of the updated DataFrame:


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao,Saldo_Balança_Comercial
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,7788,0,966,0,7788
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,1544,0,1,0,1544
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,907,0,73,0,907
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,3942,0,75,0,3942
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,29604,0,6949,0,29604


In [ ]:
dado_final.to_csv(origem + 'dado_final.csv', index=False)
print("DataFrame 'dado_final' saved to dado_final.csv in Google Drive.")

DataFrame 'dado_final' saved to dado_final.csv in Google Drive.


In [ ]:
dado_final.info()

In [ ]:
dado_final.drop_duplicates(inplace=True)
print("Duplicate rows removed.")

In [ ]:
dado_final.info()

In [ ]:
dado_final.head(5)

In [ ]:
dado_final.tail(5)

In [ ]:
coluna = 'NO_PAIS'
cp = dado_final[coluna].value_counts().sort_values(ascending=False)
print(f'total de operações {coluna}:')
print(cp)

In [ ]:
print(dado_final.columns)

In [ ]:
dado_final.to_csv(origem+'dado_final.csv',index=False)

# Task
Tag the `exp` DataFrame with 'Tipo_Operacao' as 'Export' and the `imp` DataFrame with 'Tipo_Operacao' as 'Import'. Concatenate these tagged DataFrames into `dado_final`. Then, in `dado_final`, create new columns 'VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', and 'KG_LIQUIDO_Importacao' based on the 'Tipo_Operacao' column. Calculate 'Saldo_Balança_Comercial' by subtracting 'VL_FOB_Importacao' from 'VL_FOB_Exportacao'. Finally, display the head of the updated `dado_final` DataFrame.

## Tag Export Data

### Subtask:
Add a new column, 'Tipo_Operacao', to the `exp` DataFrame with the value 'Export' for all rows.


**Reasoning**:
To add a new column 'Tipo_Operacao' with the value 'Export' to the `exp` DataFrame, I will use a direct assignment within a code block.



In [ ]:
exp['Tipo_Operacao'] = 'Export'
print("Added 'Tipo_Operacao' column to exp DataFrame with value 'Export'.")
exp.head()

Added 'Tipo_Operacao' column to exp DataFrame with value 'Export'.


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export


## Tag Import Data

### Subtask:
Add a new column, 'Tipo_Operacao', to the `imp` DataFrame with the value 'Import' for all rows.

**Reasoning**:
To add a new column 'Tipo_Operacao' with the value 'Import' to the `imp` DataFrame, I will use a direct assignment within a code block.



In [ ]:
imp['Tipo_Operacao'] = 'Import'
print("Added 'Tipo_Operacao' column to imp DataFrame with value 'Import'.")
imp.head()

Added 'Tipo_Operacao' column to imp DataFrame with value 'Import'.


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao
0,2021,7,4115,23,SP,3416200,3668,11688,4115,Couro reconstituído à base de couro ou de fibr...,Alemanha,3416200,Franca,Import
1,2021,4,8525,160,SP,3424709,1,3685,8525,Aparelhos emissores (transmissores) para radio...,China,3424709,Jaguariúna,Import
2,2021,7,8302,160,SP,3409205,5043,45918,8302,"Guarnições, ferragens e artigos semelhantes, d...",China,3409205,Cajamar,Import
3,2021,1,4009,249,SP,3420509,10417,207124,4009,"Tubos de borracha vulcanizada não endurecida, ...",Estados Unidos,3420509,Indaiatuba,Import
4,2021,11,8536,160,SP,3450308,306224,3298907,8536,"Aparelhos para interrupção, seccionamento, pro...",China,3450308,São Paulo,Import


## Concatenate DataFrames

### Subtask:
Concatenate the `exp` and `imp` DataFrames into `dado_final`, ignoring the original index.

**Reasoning**:
Both `exp` and `imp` DataFrames have now been tagged with their respective 'Tipo_Operacao'. The next logical step, as per the main task, is to concatenate these two DataFrames into `dado_final` using `pd.concat` and ignore the original index. Then, I will display the info of the `dado_final` DataFrame to verify the concatenation.



In [ ]:
dado_final = pd.concat([exp, imp], ignore_index=True)
print("DataFrames 'exp' and 'imp' concatenated into 'dado_final'.")
dado_final.info()

DataFrames 'exp' and 'imp' concatenated into 'dado_final'.
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13159539 entries, 0 to 13159538
Data columns (total 14 columns):
 #   Column         Dtype 
---  ------         ----- 
 0   CO_ANO         int64 
 1   CO_MES         int64 
 2   SH4            int64 
 3   CO_PAIS        int64 
 4   SG_UF_MUN      object
 5   CO_MUN         int64 
 6   KG_LIQUIDO     int64 
 7   VL_FOB         int64 
 8   CO_SH4         int64 
 9   NO_SH4_POR     object
 10  NO_PAIS        object
 11  CO_MUN_GEO     int64 
 12  NO_MUN_MIN     object
 13  Tipo_Operacao  object
dtypes: int64(9), object(5)
memory usage: 1.4+ GB


## Create Specific Trade Columns

### Subtask:
Create new columns in `dado_final`: 'VL_FOB_Exportacao' and 'KG_LIQUIDO_Exportacao' for 'Export' operations, and 'VL_FOB_Importacao' and 'KG_LIQUIDO_Importacao' for 'Import' operations. Initialize these new columns with 0.

**Reasoning**:
The subtask requires initializing specific trade columns in the `dado_final` DataFrame. I will create the new columns 'VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', and 'KG_LIQUIDO_Importacao' and set their initial values to 0.



In [ ]:
dado_final['VL_FOB_Exportacao'] = 0.0
dado_final['VL_FOB_Importacao'] = 0.0
dado_final['KG_LIQUIDO_Exportacao'] = 0.0
dado_final['KG_LIQUIDO_Importacao'] = 0.0
print("Initialized new trade-specific columns in 'dado_final'.")
dado_final.head()

Initialized new trade-specific columns in 'dado_final'.


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,0.0,0.0,0.0,0.0
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,0.0,0.0,0.0,0.0
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,0.0,0.0,0.0,0.0
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,0.0,0.0,0.0,0.0
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,0.0,0.0,0.0,0.0


## Populate Trade Columns

### Subtask:
Populate 'VL_FOB_Exportacao' and 'KG_LIQUIDO_Exportacao' for 'Export' operations, and 'VL_FOB_Importacao' and 'KG_LIQUIDO_Importacao' for 'Import' operations in `dado_final`.

**Reasoning**:
Now that the trade-specific columns are initialized, I will populate 'VL_FOB_Exportacao' and 'KG_LIQUIDO_Exportacao' for 'Export' operations and 'VL_FOB_Importacao' and 'KG_LIQUIDO_Importacao' for 'Import' operations using conditional assignment based on the 'Tipo_Operacao' column. After population, I will display the head of the DataFrame to verify the changes.



In [ ]:
dado_final.loc[dado_final['Tipo_Operacao'] == 'Export', 'VL_FOB_Exportacao'] = dado_final['VL_FOB']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Export', 'KG_LIQUIDO_Exportacao'] = dado_final['KG_LIQUIDO']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Import', 'VL_FOB_Importacao'] = dado_final['VL_FOB']
dado_final.loc[dado_final['Tipo_Operacao'] == 'Import', 'KG_LIQUIDO_Importacao'] = dado_final['KG_LIQUIDO']
print("Populated trade-specific columns in 'dado_final'.")
dado_final.head()

Populated trade-specific columns in 'dado_final'.


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,7788.0,0.0,966.0,0.0
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,1544.0,0.0,1.0,0.0
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,907.0,0.0,73.0,0.0
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,3942.0,0.0,75.0,0.0
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,29604.0,0.0,6949.0,0.0


## Calculate Trade Balance

### Subtask:
Calculate 'Saldo_Balança_Comercial' by subtracting 'VL_FOB_Importacao' from 'VL_FOB_Exportacao'.

**Reasoning**:
Now that the trade-specific columns are populated, I will calculate 'Saldo_Balança_Comercial' by subtracting 'VL_FOB_Importacao' from 'VL_FOB_Exportacao' and then display the head of the DataFrame to verify the new column.



In [ ]:
dado_final['Saldo_Balança_Comercial'] = dado_final['VL_FOB_Exportacao'] - dado_final['VL_FOB_Importacao']
print("Calculated 'Saldo_Balança_Comercial'.")
dado_final.head()

Calculated 'Saldo_Balança_Comercial'.


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao,Saldo_Balança_Comercial
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,7788.0,0.0,966.0,0.0,7788.0
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,1544.0,0.0,1.0,0.0,1544.0
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,907.0,0.0,73.0,0.0,907.0
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,3942.0,0.0,75.0,0.0,3942.0
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,29604.0,0.0,6949.0,0.0,29604.0


## Summary:

### Data Analysis Key Findings
*   The `exp` DataFrame was tagged with 'Tipo_Operacao' as 'Export', and the `imp` DataFrame was tagged with 'Tipo_Operacao' as 'Import'.
*   The `exp` and `imp` DataFrames were successfully concatenated into a single `dado_final` DataFrame, resulting in a dataset with 13,159,539 entries.
*   Four new columns \(`VL_FOB_Exportacao`, `VL_FOB_Importacao`, `KG_LIQUIDO_Exportacao`, `KG_LIQUIDO_Importacao`\) were created in `dado_final`. These columns were conditionally populated: `VL_FOB_Exportacao` and `KG_LIQUIDO_Exportacao` received values from 'VL_FOB' and 'KG_LIQUIDO' respectively for 'Export' operations, while `VL_FOB_Importacao` and `KG_LIQUIDO_Importacao` were populated for 'Import' operations.
*   A new column, `Saldo_Balança_Comercial`, was calculated by subtracting `VL_FOB_Importacao` from `VL_FOB_Exportacao` in the `dado_final` DataFrame.

### Insights or Next Steps
*   The consolidated `dado_final` DataFrame, with distinct export and import values and the calculated trade balance, provides a comprehensive view for analyzing trade activities.
*   The next step could involve aggregating this data by specific criteria such as product category, partner country, or time period to identify key trade patterns, surpluses, or deficits.


# Task
Remove decimal points from the value columns ('VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', 'KG_LIQUIDO_Importacao', 'Saldo_Balança_Comercial') in the `dado_final` DataFrame, handling any potential NaN values, and then save the processed `dado_final` DataFrame to a CSV file named 'dado_final.csv' in your Google Drive, overwriting any previous version. Finally, confirm that the `dado_final` DataFrame has been fully processed and saved with all requested corrections.

## Remove Decimal Points from Value Columns

### Subtask:
Convert the value columns ('VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', 'KG_LIQUIDO_Importacao', 'Saldo_Balança_Comercial') to integer type to remove decimal points, handling any potential NaN values.


**Reasoning**:
I need to convert specified value columns to integer type to remove decimal points, handling NaN values, and then verify the changes. This can be achieved by iterating through the columns, applying type conversions and NaN handling, and then displaying the DataFrame's head and info.



In [ ]:
value_columns = ['VL_FOB_Exportacao', 'VL_FOB_Importacao', 'KG_LIQUIDO_Exportacao', 'KG_LIQUIDO_Importacao', 'Saldo_Balança_Comercial']

for col in value_columns:
    # Ensure the column exists before trying to convert, and handle potential NaN values
    if col in dado_final.columns:
        dado_final[col] = dado_final[col].astype(float).fillna(0).astype(int)
    else:
        print(f"Warning: Column '{col}' not found in dado_final. Skipping type conversion for this column.")

print("Decimal points removed from value columns. Displaying head of the updated DataFrame:")
dado_final.head()

Decimal points removed from value columns. Displaying head of the updated DataFrame:


,CO_ANO,CO_MES,SH4,CO_PAIS,SG_UF_MUN,CO_MUN,KG_LIQUIDO,VL_FOB,CO_SH4,NO_SH4_POR,NO_PAIS,CO_MUN_GEO,NO_MUN_MIN,Tipo_Operacao,VL_FOB_Exportacao,VL_FOB_Importacao,KG_LIQUIDO_Exportacao,KG_LIQUIDO_Importacao,Saldo_Balança_Comercial
0,2021,3,3305,493,SP,3434401,966,7788,3305,Preparações capilares,México,3434401,Osasco,Export,7788,0,966,0,7788
1,2021,7,8419,386,SP,3409502,1,1544,8419,"Aparelhos e dispositivos, mesmo aquecidos elec...",Itália,3409502,Campinas,Export,1544,0,1,0,1544
2,2021,12,8511,63,PR,4106902,73,907,8511,Aparelhos e dispositivos elétricos de ignição ...,Argentina,4106902,Curitiba,Export,907,0,73,0,907
3,2021,7,6204,586,SC,4202909,75,3942,6204,"Fatos de saia-casaco, conjuntos, casacos, vest...",Paraguai,4202909,Brusque,Export,3942,0,75,0,3942
4,2021,8,8708,63,SP,3440002,6949,29604,8708,Partes e acessórios dos veículos automóveis da...,Argentina,3440002,Pompéia,Export,29604,0,6949,0,29604


**Reasoning**:
As per the subtask instructions, after performing the type conversions and displaying the head of the DataFrame, I need to print the data types of all columns in `dado_final` using `.info()` to confirm the type change for the selected columns.



In [ ]:
print("Displaying DataFrame information to confirm type changes:")
dado_final.info()

Displaying DataFrame information to confirm type changes:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13159539 entries, 0 to 13159538
Data columns (total 19 columns):
 #   Column                   Dtype 
---  ------                   ----- 
 0   CO_ANO                   int64 
 1   CO_MES                   int64 
 2   SH4                      int64 
 3   CO_PAIS                  int64 
 4   SG_UF_MUN                object
 5   CO_MUN                   int64 
 6   KG_LIQUIDO               int64 
 7   VL_FOB                   int64 
 8   CO_SH4                   int64 
 9   NO_SH4_POR               object
 10  NO_PAIS                  object
 11  CO_MUN_GEO               int64 
 12  NO_MUN_MIN               object
 13  Tipo_Operacao            object
 14  VL_FOB_Exportacao        int64 
 15  VL_FOB_Importacao        int64 
 16  KG_LIQUIDO_Exportacao    int64 
 17  KG_LIQUIDO_Importacao    int64 
 18  Saldo_Balança_Comercial  int64 
dtypes: int64(14), object(5)
memory usage: 1.9+ GB

**Reasoning**:
The next step in the main task is to save the processed `dado_final` DataFrame to a CSV file named 'dado_final.csv' in the specified Google Drive location. This will overwrite any previous version, ensuring the updated data is stored.



In [ ]:
dado_final.to_csv(origem + 'dado_final.csv', index=False)
print("DataFrame 'dado_final' saved to dado_final.csv in Google Drive.")

DataFrame 'dado_final' saved to dado_final.csv in Google Drive.


## Final Task

### Subtask:
Confirm that the `dado_final` DataFrame has been fully processed and saved with all requested corrections.


## Summary:

### Q&A
Yes, the `dado_final` DataFrame has been fully processed by removing decimal points from the specified value columns, handling NaN values, and has been successfully saved as 'dado_final.csv' in Google Drive with all requested corrections.

### Data Analysis Key Findings
*   Decimal points were successfully removed from the columns 'VL\_FOB\_Exportacao', 'VL\_FOB\_Importacao', 'KG\_LIQUIDO\_Exportacao', 'KG\_LIQUIDO\_Importacao', and 'Saldo\_Balança\_Comercial'. These columns were converted to integer (`int64`) data type after handling potential NaN values by filling them with `0`.
*   The updated `dado_final` DataFrame, with the corrected column types, was successfully saved to a CSV file named 'dado\_final.csv' in the specified Google Drive location.

### Insights or Next Steps
*   The `dado_final` DataFrame is now prepared for further numerical analysis or reporting, ensuring consistent integer representation for key financial and volume metrics.
